weave-js/src/components/PagePanelComponents/Home/Browse3/pages/LeaderboardPage/LeaderboardPage.tsx

```diff
@@ -220,6 +259,54 @@ export const LeaderboardPageContentInner: React.FC<
       .filter(c => c != null) as LeaderboardColumnOrderType;
   }, [workingLeaderboardValCopy, evalData]);
 
+  // Calculate the number of selected models (rows) based on selectedEvaluations
+  const selectedModelsCount = useMemo(() => {
+    if (!data || selectedEvaluations.length === 0) {
+      return 0;
+    }
+
+    // Get unique model names from selected evaluations
+    const allRecords = getAllEvaluationRecords(data);
```

Comment:
i would put `const allRecords = getAllEvaluationRecords(data);` as a memoed thing before the others b/c it can be expensive with that quad loop